pkg/health/server/server.go

```diff
@@ -186,92 +186,101 @@ func (s *Server) collectNodeConnectivityMetrics() {
 	}
 	localClusterName, localNodeName := getClusterNodeName(s.localStatus.Name)
 
+	endpointStatuses := [3]int{0, 0, 0}
+	nodeStatuses := [3]int{0, 0, 0}
+
 	for _, n := range s.connectivity.nodes {
 		if n == nil || n.Host == nil || n.Host.PrimaryAddress == nil || n.HealthEndpoint == nil || n.HealthEndpoint.PrimaryAddress == nil {
 			continue
 		}
 
-		targetClusterName, targetNodeName := getClusterNodeName(n.Name)
 		nodePathPrimaryAddress := healthClientPkg.GetHostPrimaryAddress(n)
 		nodePathSecondaryAddress := healthClientPkg.GetHostSecondaryAddresses(n)
 
 		endpointPathStatus := n.HealthEndpoint
-		isEndpointReachable := healthClientPkg.SummarizePathConnectivityStatusType(healthClientPkg.GetAllEndpointAddresses(n)) == healthClientPkg.ConnStatusReachable
-		isNodeReachable := healthClientPkg.SummarizePathConnectivityStatusType(healthClientPkg.GetAllHostAddresses(n)) == healthClientPkg.ConnStatusReachable
-
-		location := metrics.LabelLocationLocalNode
-		if targetClusterName != localClusterName {
-			location = metrics.LabelLocationRemoteInterCluster
-		} else if targetNodeName != localNodeName {
-			location = metrics.LabelLocationRemoteIntraCluster
-		}
 
-		// Aggregated status for endpoint connectivity
-		metrics.NodeConnectivityStatus.WithLabelValues(
-			localClusterName, localNodeName, targetClusterName, targetNodeName, location, metrics.LabelPeerEndpoint).
-			Set(metrics.BoolToFloat64(isEndpointReachable))
+		isEndpointReachable := healthClientPkg.SummarizePathConnectivityStatusType(healthClientPkg.GetAllEndpointAddresses(n))
+		isNodeReachable := healthClientPkg.SummarizePathConnectivityStatusType(healthClientPkg.GetAllHostAddresses(n))
 
-		// Aggregated status for node connectivity
-		metrics.NodeConnectivityStatus.WithLabelValues(
-			localClusterName, localNodeName, targetClusterName, targetNodeName, location, metrics.LabelPeerNode).
-			Set(metrics.BoolToFloat64(isNodeReachable))
+		// Aggregate connectivity statuses
+		for i := healthClientPkg.ConnStatusReachable; i <= healthClientPkg.ConnStatusUnknown; i++ {
+			endpointStatuses[i] += isEndpointReachable[i]
+			nodeStatuses[i] += isNodeReachable[i]
+		}
 
 		// HTTP endpoint primary
 		collectConnectivityMetric(endpointPathStatus.PrimaryAddress.HTTP, localClusterName, localNodeName,
-			targetClusterName, targetNodeName, endpointPathStatus.PrimaryAddress.IP,
-			location, metrics.LabelPeerEndpoint, metrics.LabelTrafficHTTP, metrics.LabelAddressTypePrimary)
```

Comment:
I think we can keep `LabelTrafficHTTP` and `LabelTrafficICMP` as they don't increase cardinality, but are probably nice to understand what is measured.

I would also keep `LabelAddressTypePrimary` and `LabelAddressTypeSecondary`. Without it, we are overriding results for "primary" address with results of "secondary" address. Also, it doesn't affect cardinality that much, only x2.